### PR TITLE
Implement ApplicationHandler for `&mut A` and `Box<A>`

### PR DESCRIPTION
**Pull request by madsmtm**
I'm not sure if there was a reason these implementations didn't exist before, but they're nice to have.

A future PR could change the signature of `run_app` from taking `&mut A` to just taking `A` (which, in turn, would allow us to make `Drop` work better).

- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users

### CODE REVIEW

File: src/application.rs [hunk 1]
```diff
@@ -223,3 +223,60 @@ pub trait ApplicationHandler<T: 'static = ()> {
         let _ = event_loop;
     }
 }
+
+impl<A: ApplicationHandler<T>, T: 'static> ApplicationHandler<T> for &mut A {
```
Comment (notgull) [inline src/application.rs:227]: Could we add a `?Sized` bound here to enable dynamic dispatch?

**Review by notgull — CHANGES_REQUESTED** [review]
I think we can also implement for `Box<A>`.

**Comment by madsmtm** [comment]
> I think we can also implement for `Box<A>`.

You're right, have done so now.

File: src/changelog/unreleased.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ changelog entry.
 ### Added
 
 - Reexport `raw-window-handle` versions 0.4 and 0.5 as `raw_window_handle_04` and `raw_window_handle_05`.
+- Implement `ApplicationHandler` for `&mut` references and boxes to something that implements `ApplicationHandler`.
```
Comment (notgull) [inline src/changelog/unreleased.md:47]: Nit: boxes => heap allocations

**Review by notgull — APPROVED** [review]

**Pull request merged**